server/src/main/java/io/crate/action/sql/Session.java

```diff
@@ -226,7 +226,7 @@ public void quickExec(String statement, ResultReceiver<?> resultReceiver, Row pa
             params,
             cursors,
             currentTransactionState,
-            planStats
+            new PlanStats(nodeCtx, txnCtx, tableStats)
```

Comment:
PlanStats has the lifecycle of the PlannerContext, therefore we could add the Transaction Context as a field.